pkgdb/include/flox/buildenv/buildenv.hh

```diff
@@ -79,8 +79,16 @@ struct RealisedPackage
 
 /* -------------------------------------------------------------------------- */
 
-/** @brief A conflict between two files with the same priority. */
-class BuildEnvFileConflictError : public FloxException
+/** @brief A conflict between two files with the same priority.
+ *
+ * This exception is thrown when we attempt to build an environment with two
+ * store paths with the same priority that contain the same file.
+ *
+ * This exception is intended to be caught by the caller and converted into a
+ * @a PackageConflict which restores the originating packages for display
```

Comment:
nonblocking suggestion: if this is the case, I wouldn't expect this class to inherit from `FloxException`, and I don't think it needs to share `EC_BUILDENV_CONFLICT` with `PackageConflict`